queries.sql

```diff
@@ -110,3 +106,81 @@ JOIN animals ON owners.id = animals.owner_id
 GROUP BY owners.full_name
 ORDER BY animal_count DESC
 LIMIT 1;
+
+-- Query Multiple Tables;
+SELECT a.name
+FROM animals a
+JOIN visits v ON a.id = v.animals_id
+JOIN vets vt ON v.vets_id = vt.id
+WHERE vt.name = 'William Tatcher'
+ORDER BY v.visiting_date DESC
+LIMIT 1;
+
+
+SELECT COUNT(DISTINCT v.animals_id) AS animal_count
+FROM visits v
+JOIN vets vt ON v.vets_id = vt.id
+WHERE vt.name = 'Stephanie Mendez';
+
+
+SELECT vt.name, s.name
+FROM vets vt
+LEFT JOIN specializations sp ON vt.id = sp.vets_id
+LEFT JOIN species s ON sp.species_id = s.id;
+
+
+SELECT a.name
+FROM animals a
+JOIN visits v ON a.id = v.animals_id
+JOIN vets vt ON v.vets_id = vt.id
+WHERE vt.name = 'Stephanie Mendez'
+  AND v.visiting_date >= '2020-04-01'
+  AND v.visiting_date <= '2020-08-30';
+
+
+SELECT a.name, COUNT(*) AS visit_count
+FROM animals a
+JOIN visits v ON a.id = v.animals_id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+
+
+SELECT a.name, MIN(v.visiting_date) AS first_visit
+FROM animals a
+JOIN visits v ON a.id = v.animals_id
+JOIN vets vt ON v.vets_id = vt.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name
+ORDER BY first_visit ASC
+LIMIT 1;
+
+
+SELECT a.name AS animal_name, v.visiting_date, vt.name AS vet_name, vt.age AS vet_age
+FROM animals a
+JOIN visits v ON a.id = v.animals_id
+JOIN vets vt ON v.vets_id = vt.id
+WHERE v.visiting_date = (
+    SELECT MAX(visiting_date)
+    FROM visits
+)
+LIMIT 1;
+
+
+SELECT COUNT(*) AS num_visits
+FROM visits v
+JOIN vets vt ON v.vets_id = vt.id
+JOIN animals a ON v.animals_id = a.id
+JOIN specializations s ON vt.id = s.vets_id AND a.id = s.species_id
+WHERE s.species_id IS NULL;
+
```

Comment:
- [ ] Great job so far @joyapisi There are just a few changes you need to make and you'll be on your way to the next project :+1: 
- [ ] I noticed that the result of your last two queries was incorrect. For the first query, even from the data you inserted, Maisy Smith is the vet without a specialization. However, she had about 9 visits from different animals. But your query brought back the image below with the number of visits as 0. Which makes it wrong. So kindly fix this :+1: 
![image](https://github.com/joyapisi/vet-clinic-db/assets/102699958/baefc014-7b7f-4cd1-afb6-b9125057e64c)